zerver/lib/typed_endpoint_validators.py

```diff
@@ -49,3 +52,31 @@ def check_url(val: str) -> str:
         return val
     except ValidationError:
         raise ValueError(_("Not a URL"))
+
+
+def to_timezone_or_empty(s: str) -> str:
+    try:
+        s = canonicalize_timezone(s)
+        zoneinfo.ZoneInfo(s)
+    except (ValueError, zoneinfo.ZoneInfoNotFoundError):
+        return ""
+    else:
+        return s
+
+
+def timezone_or_empty_validator() -> AfterValidator:
+    return AfterValidator(lambda s: to_timezone_or_empty(s))
+
+
+def to_non_negative_int_or_none(s: str) -> int | None:
+    try:
+        i = int(s)
+        if i < 0:
+            return None
+        return i
+    except ValueError:
+        return None
+
+
+def non_negative_int_or_none_validator() -> BeforeValidator:
+    return BeforeValidator(lambda s: to_non_negative_int_or_none(s))
```

Comment:
Why is this one a `BeforeValidator` rather than an `AfterValidator`?

Reply:
`to_converted_or_fallback` checked if the parameter met the required validation criteria, if not it would assign the `fallback` in this case `None`. 

Using `AfterValidator` would validate the parameter against `Int` first and then run the additional validation that is specified in the validator. What we need to do here is run the validator before and if it fails validation we return `None` then this value is checked against `Int | None` (which it will always pass) .